File: src/main/java/frc/utils/rumble/RumblePreset.java
package frc.utils.rumble;

public enum RumblePreset {
    /**
     * single pulse
    */
    TAP,
    /**
     * two taps
     */
    DOUBLE_TAP,
    /**
     * simaler to a tap, but longer
     */
    RING;

    public Rumble[] load(){
        RumblePreset rumble = this;
        switch (rumble) {
            case TAP:
                return new Rumble[]{
                    new Rumble(0.2, 1)
                };
            case RING:
                return new Rumble[]{
                    new Rumble(0.5, 0.75)
                };
            case DOUBLE_TAP:
                return new Rumble[]{
                    new Rumble(0.2, 1),
                    new Rumble(0.1, 0),
                    new Rumble(0.2, 1),
                };
        
            default:
                return new Rumble[]{};
        }
    }
}

File: src/main/kotlin/frc/utils/rumble/RumblePreset.kt
package frc.utils.rumble

enum class RumblePreset {
    /**
     * single pulse
     */
    TAP,
    /**
     * two taps
     */
    DOUBLE_TAP,
    /**
     * similar to a tap, but longer
     */
    RING;

    fun load(): Array<Rumble> {
        return when (this) {
            TAP -> arrayOf(Rumble(0.2, 1.0))
            RING -> arrayOf(Rumble(0.5, 0.75))
            DOUBLE_TAP -> arrayOf(
                Rumble(0.2, 1.0),
                Rumble(0.1, 0.0),
                Rumble(0.2, 1.0)
            )
        }
    }
}
